src/main/java/project/bot/service/TenderParserServiceBot.java

```diff
@@ -0,0 +1,78 @@
+package project.bot.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.telegram.telegrambots.bots.DefaultBotOptions;
+import org.telegram.telegrambots.bots.TelegramLongPollingBot;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.api.objects.Update;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+import project.bot.config.BotConfig;
+
+/**
+ * @author Vladyslav Pustovalov
+ * class TelegramBot which gets messages from useers and could answer to them
+ */
+@Component
+@Slf4j
+public class TenderParserServiceBot extends TelegramLongPollingBot {
+    private final BotConfig config = new BotConfig();
+    private final MessageSender messageSender = new MessageSender();
+
+    public TenderParserServiceBot(DefaultBotOptions options, String botToken) {
+        super(options, botToken);
+    }
+
+    @Override
+    public String getBotUsername() {
+        return config.getBotName();
+    }
+
+    /**
+     * Method which gets updates from users and sends to them an answer
+     */
+    @Override
+    public void onUpdateReceived(Update update) {
+        String userName = update.getMessage().getChat().getUserName();
+        String messageText = update.getMessage().getText();
+        long chatId = update.getMessage().getChatId();
+
```

Comment:
Maybe ?

```
Message message = update.getMessage();
String userName = message.getChat().getUserName();
String messageText = message.getText();
long chatId = message.getChatId();
```

Reply:
is it really necessary?)

Reply:
3 times update.getMessage() 
I think it is DRY)